packages/strapi-plugin-content-manager/admin/src/containers/EditView/index.js

```diff
@@ -289,121 +289,51 @@ function EditView({
       const filesMap = getMediaAttributes(layout, groupLayoutsData);
-      // Create the formdata to upload all the files
-      const formDatas = Object.keys(filesMap).reduce((acc, current) => {
-        const keys = current.split('.');
-        const isMultiple = get(filesMap, [current, 'multiple'], false);
-        const isGroup = get(filesMap, [current, 'isGroup'], false);
-        const isRepeatable = get(filesMap, [current, 'repeatable'], false);
-
-        const getFilesToUpload = path => {
-          const value = get(modifiedData, path, []) || [];
-
-          return value.filter(file => {
-            return file instanceof File;
-          });
-        };
-        const getFileToUpload = path => {
-          const file = get(modifiedData, [...path, 0], '');
-          if (file instanceof File) {
-            return [file];
-          }
-
-          return [];
-        };
-
-        if (!isRepeatable) {
-          const currentFilesToUpload = isMultiple
-            ? getFilesToUpload(keys)
-            : getFileToUpload([...keys]);
+      // Create an object that maps the keys with the related files to upload
+      const filesToUpload = mapDataKeysToFilesToUpload(filesMap, modifiedData);
 
-          if (!isEmpty(currentFilesToUpload)) {
-            acc[current] = currentFilesToUpload.reduce((acc2, curr) => {
-              acc2.append('files', curr);
-
-              return acc2;
-            }, new FormData());
-          }
-        }
-
-        if (isGroup && isRepeatable) {
-          const [key, targetKey] = current.split('.');
-          const groupData = get(modifiedData, [key], []);
-          const groupFiles = groupData.reduce((acc1, current, index) => {
-            const files = isMultiple
-              ? getFileToUpload([key, index, targetKey])
-              : getFileToUpload([key, index, targetKey]);
+      const cleanedData = cleanData(
+        cloneDeep(modifiedData),
+        layout,
+        groupLayoutsData
+      );
 
-            if (!isEmpty(files)) {
-              const toFormData = files.reduce((acc2, curr) => {
-                acc2.append('files', curr);
+      const formData = new FormData();
 
-                return acc2;
-              }, new FormData());
+      formData.append('data', JSON.stringify(cleanedData));
 
-              acc1[`${key}.${index}.${targetKey}`] = toFormData;
-            }
+      Object.keys(filesToUpload).forEach(key => {
+        const files = filesToUpload[key];
 
-            return acc1;
-          }, {});
+        files.forEach(file => {
+          formData.append(`files.${key}`, file);
+        });
+      });
 
-          return { ...acc, ...groupFiles };
-        }
+      // Helper to visualize the formdata
+      // for (let pair of formData.entries()) {
+      //   console.log(pair[0] + ', ' + pair[1]);
+      // }
 
-        return acc;
-      }, {});
       // Change the request helper default headers so we can pass a FormData
```

Comment:
this as well as aren't we removing formdata @alexandrebodin

Reply:
nope we aren't we just removed the old one and made a new one so people can upload with it in the generated apis too